### PR TITLE
fixed some issues

### PR DESCRIPTION
**Pull request by TopiSenpai**
if you use the gradle run command the `logs` & `application.yml` should be excluded

`sourceCompatibility` shouldprob  always be 11 as thats what lavalink is compiled with

the Test class also did not fulfill the interface anymore

### CODE REVIEW

**Comment by TopiSenpai** [comment]
including shadow to build fat jars should also be important

**Comment by TopiSenpai** [comment]
seems like it needs
https://imperceptiblethoughts.com/shadow/publishing/#publishing-with-maven-publish-plugin

**Comment by TopiSenpai** [comment]
another problem with the current setup is that plugins get big(70MB) in size due to the `com.github.freyacodes.lavalink:Lavalink-Server` dependency

File: src/main/java/org/example/plugin/Test.java [hunk 1]
```diff
@@ -8,14 +8,15 @@
...
-    public Test() {
-        log.info("Hello, world!");
-    }
+	private static final Logger log = LoggerFactory.getLogger(Test.class);
 
-    @Override
-    public void onWebSocketOpen(ISocketContext context) {
-        log.info("Websocket opened!");
-    }
+	public Test() {
```
Comment (freyacodes) [inline src/main/java/org/example/plugin/Test.java:14]: Don't convert to tabs

File: build.gradle [hunk 2]
```diff
@@ -19,7 +22,7 @@ repositories {
 
 dependencies {
     implementation "dev.arbjerg.lavalink:plugin-api:0.7.0"
-    runtimeOnly "com.github.freyacodes.lavalink:Lavalink-Server:feature~plugins-SNAPSHOT"
+    runtimeOnly "com.github.freyacodes.lavalink:Lavalink-Server:feature~plugins"
```
Comment (freyacodes) [inline build.gradle:25]: Why are you changing this?

File: build.gradle [hunk 3]
```diff
@@ -2,11 +2,14 @@ plugins {
     id 'java'
     id 'application'
     id 'maven-publish'
+    id 'com.github.johnrengelman.shadow' version '7.1.0'
 }
 
 group 'org.example'
 version '1.0'
 mainClassName = "org.springframework.boot.loader.JarLauncher"
+sourceCompatibility = 11
```
Comment (freyacodes) [inline build.gradle:11]: I think I had some issues with this in Jitpack. Does Jitpack still compile this project?

**Review by freyacodes — CHANGES_REQUESTED** [review]

**Comment by freyacodes** [comment]
> commenting out and then building is my current fix which is meh

There's probably a way of fixing this. Gradle's dependency configuration is extendable

File: build.gradle (hunk 2 above)
Comment (TopiSenpai) [reply build.gradle:25]: for me it couldn't resolve it with `-SNAPSHOT`
looking on jitpack it should hm
weird

File: src/main/java/org/example/plugin/Test.java (hunk 1 above)
Comment (TopiSenpai) [reply src/main/java/org/example/plugin/Test.java]: whopsie

File: build.gradle (hunk 2 above)
Comment (TopiSenpai) [reply build.gradle:25]: getting this with `-SNAPSHOT`
> Could not find Lavalink-Server-feature~plugins-SNAPSHOT.jar (com.github.freyacodes.lavalink:Lavalink-Server:feature~plugins-SNAPSHOT:3.0-gf5b2a17-574).

File: build.gradle (hunk 3 above)
Comment (TopiSenpai) [reply build.gradle:11]: I see...
```
* What went wrong:
Execution failed for task ':compileJava'.
> invalid source release: 11
```

Comment (TopiSenpai) [reply build.gradle:11]: apparently you need to set it to java 11 in your jitpack.yml
lemme try that
https://github.com/jitpack/jitpack.io/issues/4022

Comment (TopiSenpai) [reply build.gradle:11]: that seems to have fixed it!
[`6445d75` (#1)](https://github.com/freyacodes/lavalink-plugin-template/pull/1/commits/6445d75ebb357aad522174c04d4d07b0c1040245)

https://jitpack.io/com/github/TopiSenpai/lavalink-plugin-template/-6445d75ebb-1/build.log

**Comment by TopiSenpai** [comment]
> > commenting out and then building is my current fix which is meh
> 
> There's probably a way of fixing this. Gradle's dependency configuration is extendable

I hoped you know how 👀

**Comment by freyacodes** [comment]
> I hoped you know how

I have an idea of how to do it but completing this project and writing my report takes priority

**Review by freyacodes — APPROVED** [review]

**Pull request merged**